### PR TITLE
FF Interpolate Options

### PR DESCRIPTION
**Pull request by Asizon**
I have added the most important options of `FF Interpolate` to complement the improvements added by @ctzsnooze , a review of the help button is still necessary. It work with https://github.com/betaflight/betaflight/pull/9465 Here a example:
FFInterpolate

### CODE REVIEW

File: src/js/msp/MSPHelper.js [hunk 1]
```diff
@@ -1142,6 +1142,12 @@ MspHelper.prototype.process_data = function(dataHandler) {
 
                                         if(semver.gte(CONFIG.apiVersion, "1.42.0")) {
                                             ADVANCED_TUNING.itermRelaxCutoff = data.readU8();
+											
+                                            if(semver.gte(CONFIG.apiVersion, "1.43.0")) {
```
Comment (mikeller) [inline src/js/msp/MSPHelper.js]: This has to be 1.44.0, as this will only go into configurator after 4.2 has been released.

**Comment by Asizon** [comment]
@mikeller thanks for the clarification, I was not sure how the MSP version system worked, im going to make the changes.

File: src/js/tabs/pid_tuning.js [hunk 2]
```diff
@@ -378,14 +378,36 @@ TABS.pid_tuning.initialize = function (callback) {
...
+            $('input[name="ffBoost"]').val(ADVANCED_TUNING.ff_boost);
+
+            ffInterpolateCheck.change(function() {
+                var checked = $(this).is(':checked');
+
+                if (checked) {
+                    $('.ffInterpolateSp .suboption').show();
+                } else {
+                    $('.ffInterpolateSp .suboption').hide();
+                }
```
Comment (McGiverGim) [inline src/js/tabs/pid_tuning.js]: Is cleaner and shorter to replace the complete if/else by:
```
$('.ffInterpolateSp .suboption').toggle(checked);
```

File: src/js/tabs/pid_tuning.js [hunk 3]
```diff
@@ -378,14 +378,36 @@ TABS.pid_tuning.initialize = function (callback) {
...
-
         } else {
             $('.itermRelaxCutoff').hide();
             $('.dynamicNotch').hide();
             $('.rpmFilter').hide();
         }
 
+        if (semver.gte(CONFIG.apiVersion, "1.44.0")) {
+            // FF Interpolate
+            var ffInterpolateCheck = $('input[id="ffInterpolateSp"]');
```
Comment (McGiverGim) [inline src/js/tabs/pid_tuning.js]: As Sonar says, change `var` by `const`

File: src/js/tabs/pid_tuning.js [hunk 4]
```diff
@@ -378,14 +378,36 @@ TABS.pid_tuning.initialize = function (callback) {
...
+
+            ffInterpolateCheck.prop('checked', ADVANCED_TUNING.ff_interpolate_sp !== 0);
+            $('select[id="ffInterpolate"]').val(ADVANCED_TUNING.ff_interpolate_sp > 0 ? ADVANCED_TUNING.ff_interpolate_sp : 1);
+            $('input[name="ffSmoothFactor"]').val(ADVANCED_TUNING.ff_smooth_factor);
+            $('input[name="ffBoost"]').val(ADVANCED_TUNING.ff_boost);
+
+            ffInterpolateCheck.change(function() {
+                var checked = $(this).is(':checked');
```
Comment (McGiverGim) [inline src/js/tabs/pid_tuning.js]: As Sonar says, change `var` by `const`

**Comment by ctzsnooze** [comment]
Maybe some minor changes to the tooltip?
"The following options fine-tune feed forward from race aggressive to very smooth.<br><br> Mode: set it to N)_AVG for race, AVG_2 for race/fast freestyle, AVG_3 for HD recording, AVG_4 for HD cinematic smoothness.<br><br> Smoothness: limits the amount of change that any one incoming radio step can make, increase for smoother FF signal, decrease for more aggressive FF responses.<br><br> Boost: helps overcome motor lag with quick stick inputs, can increase jitter.  10 for HD, 20 for racers. Up to 30 may be useful for low authority quads."

**Comment by Asizon** [comment]
@McGiverGim a lot of thanks for your suggestions, `.toggle` is the best way to simplify.

**Comment by Asizon** [comment]
Force pushed after changing default value to `AVG_2` like https://github.com/betaflight/betaflight/pull/9466

**Comment by Asizon** [comment]
Im sorry, need to fix the forgotten white space 🐙

**Comment by Asizon** [comment]
Rebased and conflicts solved for 10.8.0!!

File: locales/en/messages.json [hunk 5]
```diff
@@ -1761,6 +1761,33 @@
...
+        "message": "FF Interpolate"
+    },
+    "pidTuningFfInterpolateSpHelp": {
+        "message": "The following options fine-tune feed forward from race/aggressive to smooth/HD.<br><br> Mode: set to NO_AVG for race, AVG_2 for race/fast freestyle, AVG_3 for HD recording, AVG_4 for HD cinematic smoothness.<br><br> Smoothness: limits the amount of change that any one incoming radio step can make, increase for smoother FF signal, decrease for more aggressive FF responses.<br><br> Boost: helps overcome motor lag with quick stick inputs, can increase jitter. Try 10 for HD, 20 for racers. Up to 30 may be useful for low authority quads but can cause micro overshoot or jitter."
+    },
+    "pidTuningFfInterpolate": {
+        "message": "Mode"
+    },
+    "pidTuningFfInterpolateSpOptionNoAverage": {
+        "message": "No AVG"
```
Comment (mikeller) [inline locales/en/messages.json:1774]: Should we write 'average' instead of 'AVG' for these - I think this will be a bit easier to understand.

Comment (ctzsnooze) [reply locales/en/messages.json:1774]: Better for end-users to comment, perhaps.
I think most would understand the abbreviation to AVG if space was a problem.
In the help messages, 'No averaging' might be more clear.

**Comment by Asizon** [comment]
Done @mikeller rebased and renamed to `Average`

**Comment by Asizon** [comment]
image

**Pull request merged**